src/main/java/org/sciborgs1155/lib/MotorClosedLoopController.java

```diff
@@ -0,0 +1,102 @@
+package org.sciborgs1155.lib;
+
+import static edu.wpi.first.units.Units.Seconds;
+import static org.sciborgs1155.robot.Constants.PERIOD;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile.Constraints;
+import edu.wpi.first.math.trajectory.TrapezoidProfile.State;
+import monologue.Annotations.Log;
+import monologue.Logged;
+
+public class MotorClosedLoopController implements Logged {
```

Comment:
You also don't actually do any less work - it makes the code look better on the outside, but I would argue it's harder to debug and get a grasp of since you're traveling between different files so much to process things. This could just be a me thing ngl

Reply:
There are honestly so many things that can be screwed up with PID and FFD(from personal experience). So I just like the reassurance that whatever is wrong with my code does not involve any of the methods within this class(because I do not want to touch them. ever.). And secondly(the entire reason I made this in the first place), is so I can have the graphs in advantage scope as opposed to creating fields for all of this data in drive(it's really useful for debugging). I will delete the 'goalPose' field and all it's related junk because it just doesn't really belong in this class. Like you said, it'd be better in drive.